tabular/src/autogluon/tabular/trainer/model_presets/presets.py

```diff
@@ -203,7 +204,10 @@ def get_preset_models_softclass(path, hyperparameters, num_classes=None, hyperpa
         # TODO: add support for per-stack level hyperparameters
     models = get_preset_models(path=path, problem_type=SOFTCLASS, eval_metric=soft_log_loss, stopping_metric=soft_log_loss,
                                hyperparameters=hyperparameters_standard, num_classes=num_classes, hyperparameter_tune=hyperparameter_tune,
-                                name_suffix=name_suffix, default_priorities=DEFAULT_SOFTCLASS_PRIORITY)
+                                name_suffix=name_suffix, default_priorities=DEFAULT_SOFTCLASS_PRIORITY, invalid_model_names=invalid_model_names)
+    if invalid_model_names is None:
+        invalid_model_names = []
+    invalid_model_names = invalid_model_names + [model.name for model in models]
```

Comment:
could ensure name uniqueness by also adding to `invalid_names` here the new model names after the rename that occurs later on:

`model.rename(model.name.replace('Regressor', 'Classifier'))`

Reply:
revised